### PR TITLE
Upgrade select2-rails to 4.0

### PR DESCRIPTION
**Pull request by przemosk**
TODO:
- [x] deal with removing last element i.e products in promotions form
- [x] after refresh on `customer` step in admin checkout values from fields are hidden by `clear_address_fields()` js function, check why
- [x] css styles

### CODE REVIEW

**Comment by MatthewKennedy** [comment]
@przemosk  I have a vision of the upgrader to select2 v4 -> https://github.com/MatthewKennedy/spree/tree/select2-v4

I remember battling away with this, I think I fixed it in the end.
see if the js from tis file helps: https://github.com/MatthewKennedy/spree/blob/select2-v4/backend/app/assets/javascripts/spree/backend/checkouts/edit.js

**Comment by przemosk** [comment]
> @przemosk I tried to update select2 to v4 -> https://github.com/MatthewKennedy/spree/tree/select2-v4
> 
> I remember battling away with this, I think I fixed it in the end. see if the js from tis file helps with the address fields. https://github.com/MatthewKennedy/spree/blob/select2-v4/backend/app/assets/javascripts/spree/backend/checkouts/edit.js

@MatthewKennedy will take a look, thanks ! 😄

**Comment by MatthewKennedy** [comment]
@przemosk You ma need to use some of the view files here:
 https://github.com/MatthewKennedy/spree/tree/select2-v4/backend/app/views/spree/admin/orders/customer_details

Pick through the changes and see if there is anything in there that is helpful.

File: cmd/spree_cmd.gemspec [hunk 1]
```diff
@@ -26,5 +26,5 @@ Gem::Specification.new do |s|
 
   s.add_development_dependency 'rspec'
   # Temporary hack until https://github.com/wycats/thor/issues/234 is fixed
```
Comment (poudelprakash) [inline cmd/spree_cmd.gemspec:28]: Is this comment still required?

Comment (damianlegawiec) [reply cmd/spree_cmd.gemspec:28]: @poudelprakash nah, can be removed, feel free to submit a PR with this change

**Pull request merged**